cmake/checks/check_01_compiler_features.cmake

```diff
@@ -336,3 +337,26 @@ CHECK_CXX_SOURCE_COMPILES(
   DEAL_II_COMPILER_HAS_DIAGNOSTIC_PRAGMA)
 RESET_CMAKE_REQUIRED()
 
+
+#
+# Use the 'gold' linker if possible, given that it's substantially faster.
+#
+# We have to try to link a full executable with -fuse-ld=gold to check
+# whether "ld.gold" is actually available. gcc has the bad habit of
+# accepting the flag without emitting an error.
+#
+# Wolfgang Bangerth, Matthias Maier, 2015
+#
+PUSH_CMAKE_REQUIRED("-Werror")
+PUSH_CMAKE_REQUIRED("-fuse-ld=gold")
+CHECK_CXX_SOURCE_COMPILES(
```

Comment:
I think the best solution would actually be to ensure that it doesn't just compile but also links. Is this implied?